dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -3560,24 +3562,33 @@ Sedp::TypeLookupReplyReader::~TypeLookupReplyReader()
 
 DDS::ReturnCode_t
 Sedp::TypeLookupRequestWriter::send_type_lookup_request(XTypes::TypeIdentifierSeq& type_ids,
-                                                        const DCPS::RepoId& reader,
-                                                        DCPS::SequenceNumber& sequence,
-                                                        const DCPS::SequenceNumber& rpc_sequence,
-                                                        const DCPS::RepoId& participant_id)
+  const DCPS::RepoId& reader,
+  DCPS::SequenceNumber& sequence,
+  const DCPS::SequenceNumber& rpc_sequence,
+  const DCPS::RepoId& participant_id,
+  const CORBA::ULong tl_kind)
 {
-  XTypes::TypeLookup_Request type_lookup_request;
-
-  type_lookup_request.data.getTypes.type_ids = type_ids;
-  type_lookup_request.data.kind = XTypes::TypeLookup_getTypes_HashId;
+  if (tl_kind != XTypes::TypeLookup_getTypes_HashId &&
+      tl_kind != XTypes::TypeLookup_getDependencies_HashId) {
+    return DDS::RETCODE_BAD_PARAMETER;
+  }
 
+  XTypes::TypeLookup_Request type_lookup_request;
   type_lookup_request.header.request_id.writer_guid = get_repo_id();
-
   type_lookup_request.header.request_id.sequence_number.high = rpc_sequence.getHigh();
   type_lookup_request.header.request_id.sequence_number.low = rpc_sequence.getLow();
 
   // As per chapter 7.6.3.3.4 of XTypes spec
   const OPENDDS_STRING instance_name = OPENDDS_STRING("dds.builtin.TOS.") + DCPS::to_string(participant_id);
   type_lookup_request.header.instance_name = instance_name.c_str();
+  type_lookup_request.data.kind = tl_kind;
+
+  if (tl_kind == XTypes::TypeLookup_getTypes_HashId) {
```

Comment:
I think you will have repeated code either way due to repeating the typelookuprequest code for both tl_kinds.

Reply:
Using `switch` is fine, but the checking at the beginning allows it to return right away for unsupported `tl_kind`s without going through the initialization for `type_lookup_request`. If it uses `switch`, it would be better to move the `switch` block to the beginning, before the initialization of `type_lookup_request`. But even with that, I don't think there is a clear advantage of using `switch`.